src/main/java/com/mcmoddev/lib/util/TabContainer.java

```diff
@@ -1,32 +1,51 @@
 package com.mcmoddev.lib.util;
 
-import com.mcmoddev.basemetals.data.MaterialNames;
-import com.mcmoddev.basemetals.init.Materials;
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Multimap;
 import com.mcmoddev.lib.data.Names;
+import com.mcmoddev.lib.exceptions.MaterialNotFoundException;
+import com.mcmoddev.lib.exceptions.TabNotFoundException;
 import com.mcmoddev.lib.init.MMDCreativeTab;
+import com.mcmoddev.lib.init.Materials;
 import com.mcmoddev.lib.interfaces.ITabProvider;
 import com.mcmoddev.lib.material.MMDMaterial;
 
 import net.minecraft.block.Block;
 import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
 
+/**
+ * @deprecated this is the old implementation of tabcontainer, wrapped in the tabprovider interface
+ *
+ */
+@Deprecated
 public final class TabContainer implements ITabProvider {
 	// @SkyBlade: temporary code until the tabcontainer is replaced
 	
+	private static final String TAB_NOT_FOUND = "Tab not found: ";

```

Comment:
Use of this constant for message concatenation is unnecessary with the string appended inside of the `TabNotFoundException` constructor.